src/sort.c

```diff
@@ -270,10 +268,8 @@ void sortCommandGeneric(client *c, int readonly) {
     }
 
     /* Lookup the key to sort. It must be of the right types */
-    if (!storekey)
-        sortval = lookupKeyRead(c->db,c->argv[1]);
-    else
-        sortval = lookupKeyWrite(c->db,c->argv[1]);
+    int readflags = storekey ? LOOKUP_NOSTATS | LOOKUP_NONOTIFY : LOOKUP_NONE;
+    sortval = lookupKeyReadWithFlags(c->db, c->argv[1], readflags);
```

Comment:
Is this for preserving the original logic? Are we sure we don't want to fix the code so we do notify and account stats in all cases regardless of if this is a read/write command? Maybe I need some background here.

Also perhaps just
```suggestion
    sortval = lookupKeyReadWithFlags(c->db, c->argv[1], storekey ? LOOKUP_NOSTATS | LOOKUP_NONOTIFY : LOOKUP_NONE);
```

Reply:
Yes, it's for preserving the old logic.

I used the `readflags` variable to make the code more readable (give the flags a name) and to keep the lines shorter. With your change, it's a very long line.

Reply:
> Yes, it's for preserving the old logic.
> 
Can you provide some background explaining the old logic? When do we use `lookupKeyRead` vs `lookupKeyWrite` in the current approach? When should the different flags be used? Or point me to where this has been previously discussed?

Reply:
again i think this was a bug due to an innocent mistake and that we should fix the behavior in all these calls and mention it in the release notes.

Reply:
@oranagra So exactly which flags should it be here? LOOKUP_NOSTATS, LOOKUP_NONOTIFY or none?

Reply:
i think none, i think it could be a plain lookupKeyRead.
i.e. in the past it was lookupKeyWrite because of that mess we clean up (or something that we're still missing).
but i'm quite sure the part about stats and key-miss was unintended, so if we fix both we're left with a plain LookupKeyRead.